[Simon Julian Lauw] iP

simonjulianl opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](google.com))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use
All you need to do is,

1. download it from [here](google.com).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

simonjulianl committed:
Finish level 1, add basic command parser and processor

simonjulianl committed:
reformat code and make it neater

simonjulianl committed:
fix the problems by setting final variables and suppress infinte loop

simonjulianl committed:
Add indentation during printing

simonjulianl committed:
Finish add and list commands

simonjulianl committed:
Publishing Level-2

simonjulianl committed:
add the task class

simonjulianl committed:
Publish level-3, add Task and mark as done

simonjulianl committed:
Publish Level-4, refactor arguments to map, use inheritance

simonjulianl committed:
Add string testing

simonjulianl committed:
Change invalid message

simonjulianl committed:
Publish Level-5, add exception handler and test

simonjulianl committed:
Add delete command and refactor some exceptions

simonjulianl committed:
Publish Level-6, update the test with more test cases

simonjulianl committed:
clean up : cleaning up code and optimize unused codes

simonjulianl committed:
Documentation : add javadoc to public functions, clean up some public to private functions

simonjulianl committed:
Refactoring : Change commandProcessor to commandLogicUnit in naming

simonjulianl committed:
Enhancement : change Duke class to Dude

simonjulianl committed:
Enhancement : Create a data layer using Dao and refactor the usage of taskList

simonjulianl committed:
Enhancement : Add a man to check all available commands

simonjulianl committed:
Enhancement : Change bufferedReader to scanner, move it inside CommandProcessorImpl.java

simonjulianl committed:
Enhancement : Make CommandProcessorImpl Autoclosable, so the scanner would be closed in case of exception

simonjulianl committed:
Increment: Finish level-7 for persistence data storage using a local
file

simonjulianl committed:
Increment: Finish level-7 for persistence data storage using a local
file

simonjulianl committed:
Merge branch 'branch-level-7' of https://github.com/simonjulianl/ip into branch-level-7

simonjulianl committed:
[Increment] - Finish level 7 (#2)

* Enhancement : Change bufferedReader to scanner, move it inside CommandProcessorImpl.java

* Enhancement : Make CommandProcessorImpl Autoclosable, so the scanner would be closed in case of exception

* Increment: Finish level-7 for persistence data storage using a local
file

* Increment: Finish level-7 for persistence data storage using a local
file

simonjulianl committed:
Merge branch 'master' of https://github.com/simonjulianl/ip

simonjulianl committed:
[Increment] - Finish Level 8 by adding datetime for Event and Deadline (#3)

* Increment: Finish level-8 for using LocalDateTime for some deadline and
event

* Enhancement: Add data directory to gitignore

simonjulianl committed:
Increment : Merge Level 8  (#4)

* Increment: Finish level-8 for using LocalDateTime for some deadline and
event

* Enhancement: Add data directory to gitignore

simonjulianl committed:
Enhancement:  Add data path by putting ignore file (#6)

* Enhancement: Update gitignore to just ignore data/tasks

* Enhancement: Add ignore file to create data path

simonjulianl committed:
Merge branch 'master' of https://github.com/simonjulianl/ip

simonjulianl committed:
Increment: Add and Reorganize Code to Ui, Storage, Parser and Dao

simonjulianl committed:
Merge pull request #7 from simonjulianl/branch-A-MoreOOP

Increment: Merge branch more OOP

simonjulianl committed:
Merge branch 'master' of https://github.com/simonjulianl/ip

simonjulianl committed:
Increment: Finish A-JUnit by adding test to FileListStorage, TaskDao and
 ParserImpl also with some stubs

simonjulianl committed:
Merge pull request #8 from simonjulianl/branch-A-JUnit

Increment: Finish A-JUnit by adding test to FileListStorage, TaskDao and  ParserImpl also with some stubs

simonjulianl committed:
Increment: Publish v1.0 Dude JAR

simonjulianl committed:
Increment: Finalize JavaDoc and CodingStandard

simonjulianl committed:
Increment: Finalize JavaDoc comments

simonjulianl committed:
Merge branch 'master' into branch-A-JavaDoc

simonjulianl committed:
Merge pull request #9 from simonjulianl/branch-A-JavaDoc

Increment: Finalize JavaDoc comments

simonjulianl committed:
Increment: Change some comments on Dude

simonjulianl committed:
Merge pull request #10 from simonjulianl/branch-A-CodingStandard

Increment: Finalize coding standard

simonjulianl committed:
Increment: Finish Level 9 by adding find function

simonjulianl committed:
Merge pull request #11 from simonjulianl/branch-Level-9

Increment: Finish Level 9 by adding find function

simonjulianl committed:
Increment: Update JavaDoc for Event

simonjulianl committed:
Merge pull request #12 from simonjulianl/branch-Level-8

Increment: Update JavaDoc for Event

SHEZADHASSAN22 commented on src/main/java/dao/TaskDaoImpl.java:
```diff
@@ -0,0 +1,127 @@
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
```
I think your indentation levels are too big. Should be 4 spaces.

auhc99 replied:
Yes, you can change your IDE / text editor's settings to reconfigure your tab to be of 4 spacings.

simonjulianl replied:
Yup, resolved, forgot to change tabs to spaces

SHEZADHASSAN22 commented on src/main/java/dao/TaskDaoImpl.java:
```diff
@@ -0,0 +1,127 @@
+			return tasks.size();
+		} catch (Exception e) {
+			logger.warning(e.getMessage());
+			return -1;
```
Perhaps used a named constant instead of the number.

simonjulianl replied:
Resolved

SHEZADHASSAN22 commented on src/main/java/dao/TaskDaoImpl.java:
```diff
@@ -0,0 +1,127 @@
+	
+	private boolean isInvalidIndex(int index) {
+		int size = getSize();
+		return size != -1 && (index >= getSize() || index < 0);
```
Maybe this expression can be simplified for easier understanding

auhc99 replied:
I agree, This expression might be confusing to people trying to understand your code.
Maybe you can replace it with:
```
private boolean isInvalidIndex(int index) {
    int size = getSize();
    if (size == -1) {
        return false;
    } else if (index >= size || index < 0) {
        return true;
    } else {
        return false;
    }
}
```
You should also consider moving the error checking for size out of this function.

simonjulianl replied:
Sure, but I dont think using `if-else` is a good idea since I dont see any benefit from it, ill use more variables that represent better naming instead

simonjulianl committed:
Bug: Fix create data directory if not exists

simonjulianl committed:
Merge branch 'A-Gradle'

auhc99 commented on src/main/java/Dude.java:
```diff
@@ -0,0 +1,67 @@
+				+ "|  _ \\ _   _|  _ \\____\n"
+				+ "| | | | | | | | |/ __ \\\n"
+				+ "| |_| | |_| | |_|   __/\n"
+				+ "|____/ \\__,_|____\\____|\n";
```
Should this logo be extracted out out the function and be stored as a static final constant?

simonjulianl replied:
Yeah sure, I think that's a good idea

auhc99 commented on src/main/java/dao/TaskDao.java:
```diff
@@ -0,0 +1,66 @@
+/**
+ * Data access object (dao) for the Task model.
+ */
+public interface TaskDao {
```
You may want to rename this class (and other classes containing the short form "Dao") to something more understandable.

simonjulianl replied:
I think `Dao` is fine, because I think its pretty common pattern in Java, especially in Android

simonjulianl committed:
Enhancement: Add gradle for build, shadowJar, checkstyle and test

auhc99 commented on src/main/java/parser/ParserImpl.java:
```diff
@@ -0,0 +1,225 @@
+					"description", desc,
+					"timing", timing
+			));
+		} catch (Exception e) {
```
Perhaps you can specify which Exception you are handling, and not just generalise it to all exceptions.

simonjulianl replied:
Yeah sure! Resolved

simonjulianl committed:
Enhancement: Add gradle for build, shadowJar, checkstyle and test

simonjulianl committed:
Merge pull request #13 from simonjulianl/add-gradle-support

Enhancement: Add gradle for build, shadowJar, checkstyle and test

auhc99 reviewed (commented):
Overall great job with complying with the java coding standards. I noted a few minor suggestions that you can consider. I like your use of java interfaces as it makes your code more abstract.

simonjulianl committed:
Enhancement: Add checkstyle

simonjulianl commented:
> Overall great job with complying with the java coding standards. I noted a few minor suggestions that you can consider. I like your use of java interfaces as it makes your code more abstract.

Thank you so much!

simonjulianl committed:
Enhancement: Fix based on code review

simonjulianl committed:
Merge pull request #14 from simonjulianl/branch-suggestion

Enhancement: Fix based on code review

simonjulianl committed:
Merge branch 'master' of https://github.com/simonjulianl/ip

simonjulianl committed:
Enhancement: Add javadoc inherintance for all public method

simonjulianl committed:
Enhancement: Finish Level-10 by adding JavaFX Gui

simonjulianl committed:
Merge pull request #15 from simonjulianl/branch-Level-10

Enhancement: Finish Level 10 by adding JavaFX Gui

simonjulianl committed:
Fix checkStyle and comments

simonjulianl committed:
Enhancement: Add interface for window controller to update multiple windows

simonjulianl committed:
Enhancement: Add automated gradle jar compilation

simonjulianl committed:
Enhancement: Add Java assertion to check assumption of parameters

Current functions have no assertions to remind about the program preconditions.

Adding assertions in the entire codebase to remind the developers of the
 precondition of the functions. For example, to make a DialogBox, the
 string text cannot be empty.

 Using assertions are preferred to enforce string preconditions to the
 functions and make sure to catch developer mistakes as early as
 possible.

simonjulianl committed:
Merge pull request #16 from simonjulianl/branch-A-Assertions

Enhancement: Add Java assertion to check assumption of parameters

simonjulianl committed:
Enhancement: Refactor methods, variables and argument into a class

Current functions in many classes do not implement SLAP and good
abstractions.

Implementing SLAP for better code clarity, not only that also
implementing and extracting arguments into a class.

 Refactoring is important to improve the code quality.

 However, since we are not using RxJava, We still need to pass the UI
 around in case the chatbot receiving any data asynchronously (because
 there is no flow in Java that can allow the main GUI to subscribe to
 the chatbot and allows the data to propagated upwards).

simonjulianl committed:
Enhancement: Add Javadoc for arguments and enforce checkstyle

simonjulianl committed:
Merge pull request #17 from simonjulianl/branch-A-CodeQuality

Enhancement: Refactor methods, variables and argument into a class

simonjulianl committed:
Add gradle.yml

simonjulianl committed:
Changemod gradlew

simonjulianl committed:
Merge pull request #18 from simonjulianl/A-CI

Enhancement: Create basic github action

simonjulianl committed:
Enhancement: Add reminders by adding background tasks (#19)

simonjulianl committed:
Merge branch 'master' of https://github.com/simonjulianl/ip

simonjulianl committed:
CheckStyle: Enforce checkstyle rules

simonjulianl committed:
Enhancement: Add better GUI

simonjulianl committed:
Add Ui.png and checkstyle

simonjulianl committed:
Enhancement: Add user guide and fix deadline/event bug due to localdatetime parse without time

simonjulianl committed:
Set theme jekyll-theme-hacker

simonjulianl committed:
Bug: Fix typo in readme

simonjulianl committed:
Bug: Fix readme by adding find and delete into user guide
